cmd/cli/mesh_check_upgrade.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"io"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"net/http"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+)
+
+const (
+	osmReleaseRepoOwner = "openservicemesh"
+	osmReleaseRepoName = "osm"
+)
+
+const meshCheckUpgradeDesc = `
```

Comment:
Can this be integrated into the `osm version` command instead where it displays the existing version as well as new ones available?

Reply:
Hmmm that's a better experience indeed! I'll merge the two commands together since it makes sense!